packages/lab/src/slider/internal/usePointerDown.ts

```diff
@@ -0,0 +1,108 @@
+import { RefObject, useState } from "react";
+import { SliderValue, SliderChangeHandler } from "../types";
+import { getNearestIndex, getValue, setValue } from "./utils";
+
+export function usePointerDown(
+  trackRef: RefObject<HTMLDivElement>,
+  min: number,
+  max: number,
+  step: number,
+  value: SliderValue,
+  onChange: SliderChangeHandler,
+  index: number | null,
+  activeThumb: number | undefined,
+  setActiveThumb: (index: number | undefined) => void
+) {
+  const [pointerDown, setPointerDown] = useState(false);
+
+  const onDownThumb = (index: number) => {
+    const handlePointerMove = (e: PointerEvent) => onPointerMove(e, index);
+    const handlePointerUp = (e: PointerEvent) =>
+      _onPointerUp(e, handlePointerMove, handlePointerUp);
+
+    document.addEventListener("pointermove", handlePointerMove);
```

Comment:
The bug here, that was caught by the tests is that this event isn't removed on unmount